mypy/dmypy_server.py

```diff
@@ -170,11 +170,17 @@ class Server:
     # NOTE: the instance is constructed in the parent process but
     # serve() is called in the grandchild (by daemonize()).
 
-    def __init__(self, options: Options, status_file: str, timeout: int | None = None) -> None:
+    def __init__(
+        self,
+        options: Options,
+        status_file: str,
+        timeout: int | None = None,
+        options_snapshot: dict[str, object] | None = None,
```

Comment:
It feels like it shouldn't be necessary to pass both options and options_snapshot. One or the other should be enough.

Reply:
Yeah, but then we will need something like "stable compare", which will be a tiny bit slower (this is btw the original way I did it). OTOH, then motivation is much more obvious, so let's go that way indeed.

Reply:
@svalentin Could you please test again just in case?

Reply:
Nice!
Still works:

```
PS C:\src\mypy-play> python -m mypy.dmypy run .\test.py
Success: no issues found in 1 source file
```

Reply:
For what it's worth, I think this is a much more elegant fix